src/TestEnvironment.Docker.Containers.Mongo/MongoSingleReplicaSetContainerInitializer.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace TestEnvironment.Docker.Containers.Mongo
+{
+    public class MongoSingleReplicaSetContainerInitializer : IContainerInitializer<MongoSingleReplicaSetContainer>
+    {
+        private readonly string _replicaSetName;
+
+        public MongoSingleReplicaSetContainerInitializer(string replicaSetName)
+        {
+            if (string.IsNullOrWhiteSpace(replicaSetName))
+            {
+                throw new ArgumentException("The value must be specified", nameof(replicaSetName));
+            }
+
+            _replicaSetName = replicaSetName;
+        }
+
+        public async Task<bool> Initialize(
+            MongoSingleReplicaSetContainer container,
+            CancellationToken cancellationToken)
+        {
+            if (container == null)
+            {
+                throw new ArgumentNullException(nameof(container));
+            }
+
+            var mongoClient = new MongoClient(container.GetDirectNodeConnectionString());
+
+            if (await IsInitialized(mongoClient, cancellationToken))
+            {
+                return true;
+            }
+
+            await mongoClient.GetDatabase("admin").RunCommandAsync(
+                new BsonDocumentCommand<BsonDocument>(new BsonDocument
+                {
+                    {
+                        "replSetInitiate",
+                        new BsonDocument
+                        {
+                            { "_id", _replicaSetName },
+                            {
+                                "members",
+                                new BsonArray
+                                {
+                                    new BsonDocument
+                                    {
+                                        { "_id", 0 },
+                                        {
+                                            "host",
+                                            mongoClient.Settings.Server.ToString()
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }), cancellationToken: cancellationToken);
+
+            return true;
+        }
+
+        public async Task<bool> IsInitialized(IMongoClient mongoClient, CancellationToken cancellationToken)
```

Comment:
Should it be private? Looks like workaround (and hope that in the future mongo docker container initialization issue will be solved?)

Reply:
Yes, it's definitely should be private. I'll change it. In addition, I'd say it should be preserved here due to it isn't a workaround. Mongo DB replica set can be initialized only at once and such logic is checking whether the replica set was initialized or not. It's required for correct usage with reuse container functionality